pkg/document/frontmatter.go

```diff
@@ -182,20 +182,24 @@ func (f *Frontmatter) ensureID() {
 	}
 }
 
-func (d *Document) Frontmatter() (*Frontmatter, error) {
+func (d *Document) Frontmatter() *Frontmatter {
 	d.splitSource()
 
+	d.parseFrontmatter()
+
+	return d.frontmatter
+}
+
+func (d *Document) FrontmatterError() error {
```

Comment:
Suggestion: We can have two methods: `Frontmatter() *Frontmatter` and `FrontmatterWithError() (*Frontmatter, error)`. I think it will be a bit more clear and the caller will only need to use one method when it wants to handle frontmatter's error.

```suggestion
func (d *Document) FrontmatterWithError() (*Frontmatter, error) {
```